plugins/inputs/intel_powerstat/intel_powerstat.go

```diff
@@ -1,915 +1,1218 @@
-			p.logOnce[socketID+"msr"] = nil
-			cpuID, err := p.GetCPUIDFromSocketID(socketID)
-			if err != nil {
-				p.Log.Debugf("Error while reading socket ID: %v", err)
-				return
-			}
-			actualUncoreFreq, err := p.msr.readSingleMsr(cpuID, msrUncorePerfStatusString)
-			if err != nil {
-				p.Log.Debugf("Error while reading %s: %v", msrUncorePerfStatusString, err)
-				return
-			}
-			actualUncoreFreq = (actualUncoreFreq & 0x3F) * 100
-			fields["uncore_frequency_mhz_cur"] = actualUncoreFreq
-		} else {
-			err := errors.New("Error while trying to read MSR (probably msr module was not loaded), uncore_frequency_mhz_cur metric will not be collected")
-			if val := p.logOnce[socketID+"msr"]; val == nil || val.Error() != err.Error() {
-				p.Log.Errorf("%v", err)
-				// Remember that specific error occurs for socketID to omit logging next time
-				p.logOnce[socketID+"msr"] = err
-			}
+// parsePackageMetrics takes a slice of package metrics. It ensures all metrics in the slice
+// are supported package-specific, and there are no duplicates. If metrics slice is nil, then
+// default package metrics are provided.
+func parsePackageMetrics(metrics []string) ([]string, error) {
+	if metrics == nil {
+		return []string{
+			packageCurrentPowerConsumption.String(),
+			packageCurrentDramPowerConsumption.String(),
+			packageThermalDesignPower.String(),
+		}, nil
+	}
+
+	for _, m := range metrics {
+		if !isValidPackageMetric(m) {
+			return nil, fmt.Errorf("invalid package metric specified: %q", m)
 		}
 	}
-	initMinFreq, err := p.msr.retrieveUncoreFrequency(socketID, typeFreq, "min", die)
-	if err != nil {
-		p.Log.Errorf("Error while retrieving minimum uncore frequency of the socket %s, err: %v", socketID, err)
-		return
-	}
-	initMaxFreq, err := p.msr.retrieveUncoreFrequency(socketID, typeFreq, "max", die)
-	if err != nil {
-		p.Log.Errorf("Error while retrieving maximum uncore frequency of the socket %s, err: %v", socketID, err)
-		return
-	}
 
-	tags := map[string]string{
-		"package_id": socketID,
-		"type":       typeFreq,
-		"die":        die,
+	if hasDuplicate(metrics) {
+		return nil, errors.New("package metrics contains duplicates")
 	}
-	fields["uncore_frequency_limit_mhz_min"] = initMinFreq
-	fields["uncore_frequency_limit_mhz_max"] = initMaxFreq
-
-	acc.AddGauge("powerstat_package", fields, tags)
+	return metrics, nil
 }
 
-func (p *PowerStat) addThermalDesignPowerMetric(socketID string, acc telegraf.Accumulator) {
-	maxPower, err := p.rapl.getConstraintMaxPowerWatts(socketID)
-	if err != nil {
-		p.Log.Errorf("Error while retrieving TDP of the socket %s, err: %v", socketID, err)
-		return
+// parseCPUMetrics takes a slice of CPU metrics. It ensures all metrics in the slice are
+// supported CPU-specific, and there are no duplicates.
+func parseCPUMetrics(metrics []string) ([]string, error) {
+	for _, m := range metrics {
+		if !isValidCoreMetric(m) {
+			return nil, fmt.Errorf("invalid core metric specified: %q", m)
+		}
```

Comment:
Can you add a `PrintOptionValueDeprecationNotice` for `cpu_busy_cycles` here?

Reply:
I was not aware of this mechanism, thanks for bringing this up! Done as part of commit [e2de6f6b](https://github.com/influxdata/telegraf/pull/14363/commits/e2de6f6b2577fe4e9ec7b420132b2750fe7e2654).